testerum-backend/testerum-runner/testerum-runner-junit/src/main/kotlin/com/testerum/runner/junit5/provider/JunitTestProvider.kt

```diff
@@ -0,0 +1,198 @@
+import com.testerum.runner.events.model.position.EventKey
+import com.testerum.runner_cmdline.cmdline.params.model.CmdlineParams
+import com.testerum.runner_cmdline.events.execution_listeners.junit.JUnitExecutionListener
+import com.testerum.runner_cmdline.events.execution_listeners.utils.console_output_capture.ConsoleOutputCapturer
+import com.testerum.runner_cmdline.module_di.RunnerModuleBootstrapper
+import com.testerum.runner_cmdline.runner_tree.nodes.RunnerFeatureOrTest
+import com.testerum.runner_cmdline.runner_tree.nodes.feature.RunnerFeature
+import com.testerum.runner_cmdline.runner_tree.nodes.parametrized_test.RunnerParametrizedTest
+import com.testerum.runner_cmdline.runner_tree.nodes.parametrized_test.RunnerScenario
+import com.testerum.runner_cmdline.runner_tree.nodes.suite.RunnerSuite
+import com.testerum.runner_cmdline.runner_tree.nodes.test.RunnerTest
+import com.testerum.runner.junit5.logger.JUnitEventLogger
+import com.testerum_api.testerum_steps_api.test_context.ExecutionStatus
+import org.junit.jupiter.api.DynamicContainer.dynamicContainer
+import org.junit.jupiter.api.DynamicNode
+import org.junit.jupiter.api.DynamicTest.dynamicTest
+import org.junit.jupiter.api.fail
+import java.nio.file.Path
+import java.time.LocalDateTime
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.LinkedBlockingDeque
+import kotlin.concurrent.thread
+
+class JunitTestProvider(repositoryDirectory: Path,
+                        variablesEnvironment: String? = null,
+                        variableOverrides: Map<String, String> = emptyMap(),
+                        settingsFile: Path? = null,
+                        settingOverrides: Map<String, String> = emptyMap(),
+                        testPaths: List<TestPath> = emptyList(),
+                        tagsToInclude: List<String> = emptyList(),
+                        tagsToExclude: List<String> = emptyList()
+) {
+
+    private val bootstrapper: RunnerModuleBootstrapper
+    private val cmdlineParams: CmdlineParams = CmdlineParams(
+            verbose = false,
+            repositoryDirectory = repositoryDirectory,
+            variablesEnvironment = variablesEnvironment,
+            variableOverrides = variableOverrides,
+            settingsFile = settingsFile,
+            settingOverrides = settingOverrides,
+            testPaths = testPaths,
+            tagsToInclude = tagsToInclude,
+            tagsToExclude = tagsToExclude,
+            reportsWithProperties = listOf("JUNIT"),
+            managedReportsDir = null,
+            executionName = null
+    )
+
+    private lateinit var testSuiteToBeExecuted: RunnerSuite
+    private val loggingListener = JUnitEventLogger()
+
+    private lateinit var eventQueue: LinkedBlockingDeque<RunnerEvent>
+    private val eventQueueInitialized = CountDownLatch(1)
+
+    private var isTestStarted = false
+
+    init {
+        bootstrapper = RunnerModuleBootstrapper(cmdlineParams, StopWatch.start())
+    }
```

Comment:
Thre is no need for the ``init`` block. Just declare the ``bootstrapper`` property under ``cmdlineParams`` and initialize it there.